crates/core/src/update.rs

```diff
@@ -44,7 +44,7 @@ pub async fn update_dependencies(
             #[cfg(feature = "cli")]
             let p = progress.clone();
 
-            let lock = locks.iter().find(|l| l.name() == dep.name()).cloned();
+            let lock = locks.par_iter().find_any(|l| l.name() == dep.name()).cloned();
```

Comment:
Here we need `find_first` since the locks are sorted. At the moment we can't have multiple versions of the same dep but it might be the case in the future. Then the finding order becomes important.